src/precice/bindings/matlab/+precice/@SolverInterface/SolverInterface.m

```diff
@@ -0,0 +1,283 @@
+        end
+        
+        % isCouplingOngoing
+        function bool = isCouplingOngoing(obj)
+            bool = preciceGateway(uint8(21));
+        end
+        
+        % isReadDataAvailable
+        function bool = isReadDataAvailable(obj)
+            bool = preciceGateway(uint8(22));
+        end
+        
+        % isWriteDataRequired
+        function bool = isWriteDataRequired(obj,dt)
+            bool = preciceGateway(uint8(23),dt);
+        end
+        
+        % isTimestepComplete
+        function bool = isTimestepComplete(obj)
+            bool = preciceGateway(uint8(24));
+        end
+        
+        % hasToEvaluateSurrogateModel
+        function bool = hasToEvaluateSurrogateModel(obj)
+            bool = preciceGateway(uint8(25));
+        end
+        
+        % hasToEvaluateFineModel
+        function bool = hasToEvaluateFineModel(obj)
+            bool = preciceGateway(uint8(25));
+        end
+        
+        %% Action Methods
+        % isActionRequired
+        function bool = isActionRequired(obj,action)
+            if ischar(action)
+                action = string(action);
+            end
+            bool = preciceGateway(uint8(30),action);
+        end
+        
+        % fulfilledAction
+        function fulfilledAction(obj,action)
+            if ischar(action)
+                action = string(action);
+            end
+            preciceGateway(uint8(31),action);
+        end
+        
+        %% Mesh Access
+        % hasMesh
+        function bool = hasMesh(obj,meshName)
+            if ischar(meshName)
+                meshName = string(meshName);
+            end
+            bool = preciceGateway(uint8(40),meshName);
+        end
+        
+        % getMeshID
+        function id = getMeshID(obj,meshName)
+            if ischar(meshName)
+                meshName = string(meshName);
+            end
+            id = preciceGateway(uint8(41),meshName);
+        end
+        
+        % getMeshIDs
+        function ids = getMeshIDs(obj)
+            ids = preciceGateway(uint8(42));
+        end
+        
+        % getMeshHandle not yet implemented
+        
+        % setMeshVertex
+        function vertexId = setMeshVertex(obj,meshID,position)
+            vertexId = preciceGateway(uint8(44),int32(meshID),position);
+        end
+        
+        % getMeshVertexSize
+        function vertexId = getMeshVertexSize(obj,meshID)
+            vertexId = preciceGateway(uint8(45),int32(meshID));
+        end
+        
+        % setMeshVertices
+        function vertexIds = setMeshVertices(obj,meshID,inSize,positions)
+            if size(positions,2) ~= inSize
+                error('Number of columns in position vector must match size!');
+            end
+            vertexIds = preciceGateway(uint8(46),int32(meshID),int32(inSize),positions);
+        end
+        
+        % getMeshVertices
+        function positions = getMeshVertices(obj,meshID,inSize,vertexIds)
```

Comment:
We could try to get rid of `inSize` by directly extracting the size from `size(vertexIds)`.